File: homeworkfront/lib/Product.dart
class Product {
  final int productId;
  final String productName;
  final double price;
  final String category;
  final String imageurl;

  Product({
    required this.productId,
    required this.productName,
    required this.price,
    required this.category,
    required this.imageurl,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      productId: json['productid'],
      productName: json['productname'],
      price: json['price'],
      category: json['category'],
      imageurl: json['imageurl'],
    );
  }
}

File: homeworkback/homework/src/main/java/com/example/homework/Product.java
package com.example.homework;

import jakarta.persistence.*;

@Entity
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer productid;
    private String productname;
    private Double price;
    private String category;

    private String imageurl;

    public Integer getProductid() {
        return productid;
    }

    public void setProductid(Integer productid) {
        this.productid = productid;
    }

    public String getProductname() {
        return productname;
    }

    public void setProductname(String productname) {
        this.productname = productname;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getImageurl() {
        return imageurl;
    }

    public void setImageUrl(String imageurl) {
        this.imageurl = imageurl;
    }
}
